### PR TITLE
feat(staking): multi currency

### PR DESCRIPTION
**Pull request by nduchak**
adjust tests

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -2,46 +2,56 @@ pragma solidity 0.6.2;
...
 
     using SafeMath for uint256;
 
     StorageManager storageManager;
-    // amount of token staked per address
-    mapping(address => uint256) internal amountStaked;
-    // total amount of token staked per address
-    uint256 internal _totalStaked;
+    // amount of tokens per token staked per address [account -> (tokenAddress -> amount)]
+    mapping(address => mapping(address => uint256)) internal amountStaked;
```
Comment (julianmrodri) [inline contracts/Staking.sol:17]: For convention I think we should use `_amountStaked` for the naming as it is `internal`

File: contracts/Staking.sol [hunk 2]
```diff
@@ -2,46 +2,56 @@ pragma solidity 0.6.2;
...
 /// @title staking
 /// @author see ERC900 proposal
 /// @author Rinke Hendriksen <rinke@iovlabs.org>
 /// @notice implements the ERC900 interface, with some small modifications. The contract also handles native tokens.
-contract Staking {
+contract Staking is Ownable {
 
     using SafeMath for uint256;
 
     StorageManager storageManager;
```
Comment (julianmrodri) [inline contracts/Staking.sol:15]:
```suggestion
    StorageManager public storageManager;
```

Comment (julianmrodri) [reply contracts/Staking.sol:15]: I would make the `storageManager` public and lets create `function setStorageManager(address) public onlyOwner` so if for some reason we migrate to another Storage contract we can redirect the Staking to that contract.

File: contracts/Staking.sol [hunk 3]
```diff
@@ -50,52 +60,63 @@ contract Staking {
...
+    @param tokenAddress Token address
     @param data should be disregarded for the current deployment
      */
-    function stakeFor(uint256 amount, address user, bytes memory data) public payable {
+    function stakeFor(uint256 amount, address user, address tokenAddress, bytes memory data) public payable {
         // disregard passed-in amount
-        if(contractUsesNativeToken()) {
+        if(_isNativeToken(tokenAddress)) {
             amount = msg.value;
+            tokenAddress = address(0);
```
Comment (julianmrodri) [inline contracts/Staking.sol:70]: I think we should also allow to configure if the Native token is accepted. Here it is assumed you can always stake in RBTC. The problem is we cannot use the Whitelist so we would need to create a flag, and a setter function like we did [here](https://github.com/rsksmart/rif-marketplace-nfts/blob/8be8c5ad0f5c49c9f619dd47d31c5864fc03d986/contracts/ERC721SimplePlacementsV1.sol#L43). We can name it `isNativeTokenAllowed` or something on those lines.

**Review by julianmrodri — CHANGES_REQUESTED** [review]
Nice work! Added some comments on changes that I think we should do but looks good!

File: contracts/Staking.sol [hunk 4]
```diff
@@ -2,46 +2,56 @@ pragma solidity 0.6.2;
 
 import "./StorageManager.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
[the remark this answers is not in the document]
Comment (nduchak) [reply contracts/Staking.sol:4]: Done

File: contracts/Staking.sol (hunk 2 above)
Comment (nduchak) [reply contracts/Staking.sol:15]: Done

Comment (nduchak) [reply contracts/Staking.sol:15]: Nice point 👍

**Review by julianmrodri — APPROVED** [review]
LGTM Great job

**Pull request merged**